### DISCUSSION TITLE
Error: You cannot sign hex strings using sign message

### DISCUSSION THREAD

**Discussion by matt-user** [phantom/docs#189] [Help & Support] [answered]
### Summary

When passing in a Uint8Array representing a 32 byte hex string the Phantom wallet is unable to sign it as a message and throws `Error: You cannot sign solana transactions using sign message.`

### Example

https://github.com/matt-user/sign_message_repro/tree/main/my-app

### Steps to Reproduce

If you go to my github repro in the my-app folder run: npm install, npm start, and then click sign message.  I also use tweetnacl to show that is possible to sign this uint8array.

![Screenshot 2024-03-01 at 1 59 26 PM](https://github.com/phantom/docs/assets/54373384/19c087c6-2162-4f53-b7de-7b3d91f6c313)
![Screenshot 2024-03-01 at 1 59 58 PM](https://github.com/phantom/docs/assets/54373384/6115a171-92af-4124-883b-72dcbd430b47)

### Phantom Version

24.3.0

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment]
Hey @matt-user, apologies for the delayed response. Can you give me a better idea of what is encoded in this string?

**Reply by matt-user** [reply]
Hi @adamdelphantom, thanks for the response.  The hex string is a transaction id of a transaction on the Fuel network.  But does it matter what is encoded in this string?  Shouldn't the wallet be able to sign any arbitrary valid hex string, such as the tweetnacl library is able to do?

**Reply by bfriel** [reply, marked as answer]
Phantom's `signMessage` function on its Solana provider will always check if the input could be decoded as a Solana transaction. If it can be, Phantom will reject with this error message. We do this to protect users from malicious Solana dapps that try and pass transactions to `signMessage`, when they should be using `signAndSendTransaction` instead.

Solana does not currently support any initiatives like [EIP-2612](https://eips.ethereum.org/EIPS/eip-2612) that delegate spending with just a signed message. If you're looking for something like that, I would suggest leveraging our [Ethereum provider](https://docs.phantom.app/ethereum-and-polygon/signing-a-message) instead.

**Reply by xgreenx** [reply]
But it seems `signMessage` even returns the error when you try to sign one byte. It seems you require the user of the `signMessage` to encode the message in UTF8 format. Is that possible to bypass this and provide an non-utf8 bytes?
